### PR TITLE
[@dagster-io/eslint-config] Create shared config

### PR DESCRIPTION
**Pull request by hellendag**
## Summary

Create a shared eslint config for use across Dagster apps, and use it in `dagit` packages.

This also includes a `prettier` upgrade, which triggered some reformatting.

## Test Plan

Buildkite.

### CODE REVIEW

File: js_modules/dagit/packages/core/src/schedules/SchedulesTable.tsx [hunk 1]
```diff
@@ -131,14 +131,8 @@ const ScheduleRow: React.FC<{
...
-  const {
-    name,
-    cronSchedule,
-    executionTimezone,
-    futureTicks,
-    pipelineName,
-    scheduleState,
-  } = schedule;
+  const {name, cronSchedule, executionTimezone, futureTicks, pipelineName, scheduleState} =
+    schedule;
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/schedules/SchedulesTable.tsx:135]: This is a bit of a weird layout of this - new prettier?

File: js_modules/dagit/packages/eslint-config/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "@typescript-eslint/parser": "5.8.0",
+    "eslint-config-prettier": "8.3.0",
+    "eslint-plugin-graphql": "^4.0.0",
+    "eslint-plugin-import": "2.26.0",
+    "eslint-plugin-jsx-a11y": "6.5.1",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "7.27.1",
+    "eslint-plugin-react-hooks": "4.3.0",
+    "graphql": "^15.5.0",
+    "prettier": "2.6.2"
```
Comment (bengotow) [inline js_modules/dagit/packages/eslint-config/package.json:22]: I wonder if `prettier` should technically a peer dep? I guess it's not required that it's used in the parent project, but it'd be nice if the version was loose here (eg: 2.x.x) and specified in the parent so we don't get weird alignment issues if we upgrade it?

**Review by bengotow — APPROVED** [review]
LGTM!

File: js_modules/dagit/packages/core/src/schedules/SchedulesTable.tsx (hunk 1 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/schedules/SchedulesTable.tsx:135]: Yeah. I don't know why they did this.

File: js_modules/dagit/packages/eslint-config/package.json (hunk 2 above)
Comment (hellendag) [reply js_modules/dagit/packages/eslint-config/package.json:22]: Yeah, not a bad idea.

**Comment by hellendag** [comment]
Going to make `prettier@2.2.1` the peer dep, and back out the prettier upgrade. That should keep this PR a little cleaner.

**Pull request merged**